contrib/config_option_database/utils/ConfigOptions.py

```diff
@@ -0,0 +1,72 @@
+#############################################################################
+# Copyright (c) 2019 Balabit
+#
+# This program is free software; you can redistribute it and/or modify it
+# under the terms of the GNU General Public License version 2 as published
+# by the Free Software Foundation, or (at your option) any later version.
+#
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with this program; if not, write to the Free Software
+# Foundation, Inc., 51 Franklin St, Fifth Floor, Boston, MA  02110-1301  USA
+#
+# As an additional exemption you are allowed to compile & link against the
+# OpenSSL libraries as published by the OpenSSL project. See the file
+# COPYING for details.
+#
+#############################################################################
+
+from tempfile import NamedTemporaryFile
+
+from utils.BisonGraph import BisonGraph
+from utils.MergeYm import merge_grammars
+from utils.OptionParser import path_to_options
+
+
+def _make_types_terminal(graph):
+    types = [
+        'path',
+        'string',
+        'string_list',
+        'string_or_number',
+        'yesno'
+    ]
+    helpers = [
+        'filter_content',
+        'parser_content'
+    ]
+
+    for node in types + helpers:
+        graph.make_terminal(node)
+
+
+def _remove_code_blocks(graph):
+    for node in filter(lambda x: x.startswith('$@'), graph.get_nodes()):
+        graph.remove(node)
+
+
+def _remove_ifdef(graph):
+    nodes = ['KW_IFDEF', 'KW_ENDIF']
+    for node in nodes:
+        for parent in graph.get_parents(node):
+            graph.remove(parent)
+
+
+def get_driver_options():
```

Comment:
I think the scope of this function is more than only to getting "driver" "options".
This function could explore not just: drivers, but all grammar elements: junctions, channel, parsers, rewrite rules, so every statement which connects to grammar.
And explores not just options, but also blocks (I mean the whole grammar hierarchy).
Please consider to rename this function.

Reply:
I don't plan on handling the junctions or channels here. Parsers, rewrites will come here eventually, so we can change the "driver" in the function name. Any suggestions?

Reply:
What about `get_options_for_statements` or `get_options_by_statements` ?

Reply:
What about `get_options_by_statements_and_blocks` ?